artcommon/artcommonlib/konflux/konflux_db.py

```diff
@@ -86,8 +86,21 @@ def add_builds(self, builds: typing.List[KonfluxRecord], group: typing.Optional[
                 # Index by name
                 group_cache['by_name'][build.name].append(build)
 
-                # Index by NVR
-                group_cache['by_nvr'][build.nvr] = build
+                # Index by NVR - prioritize successful builds over failed ones
+                existing_build = group_cache['by_nvr'].get(build.nvr)
+                if existing_build is None:
+                    # No existing build with this NVR, add it
+                    group_cache['by_nvr'][build.nvr] = build
+                elif (
+                    existing_build.outcome != KonfluxBuildOutcome.SUCCESS
+                    and build.outcome == KonfluxBuildOutcome.SUCCESS
+                ):
+                    # Existing build is not successful, but new build is - replace with successful build
+                    group_cache['by_nvr'][build.nvr] = build
+                elif existing_build.outcome == build.outcome:
+                    # Same outcome - keep the one with the latest start_time
+                    if build.start_time and existing_build.start_time and build.start_time > existing_build.start_time:
```

Comment:
```suggestion
                elif existing_build.outcome == build.outcome and build.start_time > existing_build.start_time:
```
since `start_time` should never be None (we set it to `datetime.now(tz=timezone.utc)` in the image builder)